src/EnergyPlus/DXCoils.cc

```diff
@@ -12569,10 +12575,13 @@ namespace DXCoils {
                 // Waste heat
                 if (DXCoil(DXCoilNum).MSHPHeatRecActive) {
                     if (DXCoil(DXCoilNum).MSWasteHeat(SpeedNum) == 0) {
-                        MSHPWasteHeat = DXCoil(DXCoilNum).MSWasteHeatFrac(SpeedNum) * DXCoil(DXCoilNum).ElecCoolingPower;
+                        DXCoil(DXCoilNum).MSFuelWasteHeat = DXCoil(DXCoilNum).MSWasteHeatFrac(SpeedNum) * DXCoil(DXCoilNum).ElecCoolingPower;
                     } else {
-                        MSHPWasteHeat = CurveValue(DXCoil(DXCoilNum).MSWasteHeat(SpeedNum), OutdoorDryBulb, InletAirDryBulbTemp) *
-                                        DXCoil(DXCoilNum).MSWasteHeatFrac(SpeedNum) * DXCoil(DXCoilNum).ElecCoolingPower;
+                        DXCoil(DXCoilNum).MSFuelWasteHeat = CurveValue(DXCoil(DXCoilNum).MSWasteHeat(SpeedNum), OutdoorDryBulb, InletAirDryBulbTemp) *
+                                                            DXCoil(DXCoilNum).MSWasteHeatFrac(SpeedNum) * DXCoil(DXCoilNum).ElecCoolingPower;
```

Comment:
If waste heat is not active, then this curve may not be defined. I don't see any code in the PR to check this and throw a warning.
And I don't see any code to make sure `MSFuelWasteHeat` gets set to zero if the unit is not running.
It's hard to follow these changes here in the github diff. I'll have to look at this in more depth.

Reply:
@mjwitte I didn't realize that CoilDX_MultiSpeedCooling can only be referenced by a AirLoopHVAC:UnitaryHeatPump:AirToAir:Multispeed object. I'll revert this piece back.

Reply:
It can also be used by `AirLoopHVAC:UnitarySystem`. I'm not following why you would revert this.

Reply:
Originally the waste heat is only calculated when `DXCoil(DXCoilNum).MSHPHeatRecActive`, I thought it only applies to HP cases, so I modified it as - calculating the waste heat when 
`DXCoil(DXCoilNum).FuelType != FuelTypeElectricity` and waste heat is defined.

But since `DXCoil(DXCoilNum).MSHPHeatRecActive` is also triggered in AirLoopHVAC:UnitaryHeatPump:AirToAir:Multispeed and AirLoopHVAC:UnitarySystem, and those are the only cases apply, using the original logic should be fine.